core/station/impl/src/repositories/indexes/request_policy_resource_index.rs

```diff
@@ -404,21 +402,125 @@ mod tests {
             assert_eq!(policies.len(), 1);
             assert_eq!(policies[0], expected_method_id);
 
+            let validation_method = if i % 2 == 0 {
+                ValidationMethodResourceTarget::No
+            } else {
+                ValidationMethodResourceTarget::ValidationMethod(CanisterMethod {
+                    canister_id: Principal::management_canister(),
+                    method_name: format!("validation_method_{}", i),
+                })
+            };
+
             // and then find by validation method
             let policies = repository
                 .find_external_canister_call_policies_by_execution_and_validation_method(
                     &Principal::management_canister(),
                     &format!("method_{}", i),
-                    &ValidationMethodResourceTarget::ValidationMethod(CanisterMethod {
-                        canister_id: Principal::management_canister(),
-                        method_name: format!("validation_method_{}", i),
-                    }),
+                    &validation_method,
                 );
 
             assert_eq!(policies.len(), 1);
             assert_eq!(policies[0], expected_method_id);
         }
 
         assert!(expected_method_ids.is_empty());
+
+        // then insert another policy with the same execution method but different validation method
```

Comment:
could we also add a policy with no validation (to test that no validation is sorted before some validation and take_while works as expected)?